src/components/PermissionGuard/index.tsx

```diff
@@ -0,0 +1,24 @@
+import { useAccount } from "@containers/AccountProvider";
+import { AccessDenied } from "@components/ErrorBoundary";
+
+type PermissionGuardProps = {
+  children: JSX.Element
+  permissions: string[]
+  fallback?: JSX.Element
+}
+
+const ROLE_PREFIX = "reaction:legacy";
```

Comment:
I assume you define all permissions excluding this role prefix for simplicity and cleanliness' sake. Although it looks better without it, we prevent developers who implement custom roles to leverage this route guard. If a developer has the custom role `merchstack:current:stores/read`, we will append the reaction prefix and the permission check will fail. I would prefer all roles to be as is. What's your opinion on that?

Reply:
Yeah, that's a valid point. I will keep the role names as is